src/plugins/intel_gpu/src/kernel_selector/cl_kernels/permute_bfzyx_to_bfyxz.cl

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/fetch_data.cl"
+#include "include/batch_headers/data_types.cl"
+
+#define unroll_for __attribute__((opencl_unroll_hint)) for
+#define INPUT0_GET_TILED_INDEX(ORDER) INPUT0_GET_INDEX(ORDER)
+#define OUTPUT_GET_TILED_INDEX(ORDER) OUTPUT_GET_INDEX(ORDER)
+KERNEL (permute_bfzyx_to_bfyxz)(
+    const __global INPUT0_TYPE* input,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+    )
+{
+    const uint x = get_global_id(0);
+    const uint y = get_global_id(1);
+    const uint z = get_global_id(2) % NZ_TILES;
+    const uint f = get_global_id(2) / NZ_TILES;
+    const uint b = get_global_id(2) / (INPUT0_FEATURE_NUM * NZ_TILES);
+
+    __local OUTPUTVTYPE transpose_buf[TRANS_BUF_SIZE];
```

Comment:
The lws is {5, 1, 66} for {1, 3, 85, 20, 20} shape so there are 330 work items in the work group which can run in parallel. Wouldn't be there races between filling and reading of the transpose buffer?

Reply:
It shouldn't be. Each thread populates and reads it's own region of the local memory.